src/main/scala/viper/silver/plugin/standard/reasoning/analysis/VarAnalysisGraphMap.scala

```diff
@@ -0,0 +1,390 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2024 ETH Zurich.
+
+package viper.silver.plugin.standard.reasoning.analysis
+
+import viper.silver.ast.utility.Expressions
+import viper.silver.ast.{AccessPredicate, Apply, Assert, Assume, BinExp, CurrentPerm, Declaration, DomainFuncApp, Exhale, Exp, FieldAccess, FieldAssign, Fold, ForPerm, FuncApp, Goto, If, Inhale, Label, LocalVar, LocalVarAssign, LocalVarDecl, Method, MethodCall, Package, Position, Program, Quasihavoc, Quasihavocall, Ref, Scope, Seqn, Stmt, UnExp, Unfold, While}
+import viper.silver.plugin.standard.reasoning.analysis.AnalysisUtils.{AssumeNode, getLocalVarDeclFromLocalVar, heapVertex}
+import viper.silver.plugin.standard.reasoning.{Assumes, ExistentialElim, FlowAnnotation, FlowVar, Heap, NoAssumeAnnotation, OldCall, UniversalIntro, Var}
+import viper.silver.plugin.standard.termination.{DecreasesSpecification, DecreasesStar, DecreasesTuple, DecreasesWildcard}
+import viper.silver.verifier.{AbstractError, ConsistencyError}
+
+import scala.collection.mutable
+
+object AnalysisUtils {
+
+  // TODO Add NoAssume Specification (default for all cases except for non-terminating abstract methods (e.g AssumeFree).
+  //      Also check method Body for assumes (or inhales in case this specification is present
+  case class AssumeNode(pos: Position) extends Declaration {
+    override def name: String = ".assume"
+  }
+
+  val heapVertex: LocalVarDecl = LocalVarDecl(".heap", Ref)()
+
+  def getDeclarationFromFlowVar(f: FlowVar, m: Method): Declaration = {
+    f match {
+      case v: Var => LocalVarDecl(v.decl.name, v.decl.typ)(v.decl.pos)
+      case _: Heap => heapVertex
+      case _: Assumes => AssumeNode(m.pos)
+    }
+  }
+
+  def getLocalVarDeclFromFlowVar(f: FlowVar): LocalVarDecl = {
+    f match {
+      case v: Var => LocalVarDecl(v.decl.name, v.decl.typ)(v.decl.pos)
+      case _: Heap => heapVertex
+    }
+  }
+
+  def getLocalVarDeclFromLocalVar(l: LocalVar): LocalVarDecl = {
+    LocalVarDecl(l.name, l.typ)()
+  }
+
+  def containsDecreasesAnnotations(m: Method): (Boolean, Boolean) = {
```

Comment:
The current implementation does not cover the case where one (or more) `DecreasesTuple` are used but the associated conditions do not cover all cases. While it's fine to be conservative and just reject conditional termination measures, this function should take this into account.
Furthermore, since termination is the main objective of this function, I'd use the first return parameter to indicate that the specification (soundly!) enforces termination and only the second one to provide more insights, i.e., whether the user specified any termination specification.
Please also add a short documentation to this function explaining what is checked
```suggestion
  def specifiesTermination(m: Method): (Boolean, Boolean) = {
```